src/lls_model.jl

```diff
@@ -46,50 +178,113 @@ function LLSModel(A :: AbstractMatrix, b :: AbstractVector;
     I = ((i,j) for i = 1:m, j = 1:n)
     getindex.(I, 1)[:], getindex.(I, 2)[:], C[:]
   end
-  LLSModel(Arows, Acols, Avals, nvar, b, x0=x0, lvar=lvar, uvar=uvar,
-           Crows=Crows, Ccols=Ccols, Cvals=Cvals, lcon=lcon, ucon=ucon, y0=y0, name=name)
+  LLSTripletModel(Arows, Acols, Avals, nvar, b, x0=x0, lvar=lvar, uvar=uvar,
+                  Crows=Crows, Ccols=Ccols, Cvals=Cvals, lcon=lcon, ucon=ucon, y0=y0, name=name)
+end
+
+function LLSModel(A :: LinearOperator, b :: AbstractVector; kwargs...)
+  LLSOperatorModel(A, b; kwargs...)
 end
 
 function LLSModel(Arows :: AbstractVector{<: Integer},
+                  Acols :: AbstractVector{<: Integer},
+                  Avals :: AbstractVector,
+                  nvar :: Integer,
+                  b :: AbstractVector;
+                  variant=:triplet, kwargs...)
+  if !(variant in [:matrix, :operator, :triplet])
+    error("variant should be one of [:matrix, :operator, :triplet")
+  end
+  LLSModel(Val(variant), Arows, Acols, Avals, nvar, b; kwargs...)
+end
+
+function LLSModel(::Val{:matrix},
+                  Arows :: AbstractVector{<: Integer},
+                  Acols :: AbstractVector{<: Integer},
+                  Avals :: AbstractVector,
+                  nvar :: Integer,
+                  b :: AbstractVector;
+                  x0 :: AbstractVector = zeros(eltype(Avals), nvar),
+                  lvar :: AbstractVector = fill(-eltype(Avals)(Inf), nvar),
+                  uvar :: AbstractVector = fill(eltype(Avals)(Inf), nvar),
+                  Crows :: AbstractVector{<: Integer} = eltype(Arows)[],
+                  Ccols :: AbstractVector{<: Integer} = eltype(Arows)[],
+                  Cvals :: AbstractVector = eltype(Avals)[],
+                  lcon :: AbstractVector = eltype(Avals)[],
+                  ucon :: AbstractVector = eltype(Avals)[],
+                  y0 :: AbstractVector = zeros(eltype(Avals), length(lcon)),
+                  name :: String = "generic-LLSModel"
+                 )
+  nequ = length(b)
+  ncon = length(lcon)
+  A = sparse(Arows, Acols, Avals, nequ, nvar)
+  C = sparse(Crows, Ccols, Cvals, ncon, nvar)
+
+  LLSMatrixModel(A, b, x0=x0, lvar=lvar, uvar=uvar,
+                 C=C, lcon=lcon, ucon=ucon, y0=y0, name=name)
+end
+
+function LLSModel(::Val{:operator},
+                  Arows :: AbstractVector{<: Integer},
```

Comment:
I feel you're making your like complicated here. Maybe it would be simpler if LinearOperators defined an operator from a matrix in COO format?

Reply:
It would be simpler, and I think after the COO matrix implementation, I'll update to use it here.